runtime/src/bank.rs

```diff
@@ -4103,15 +4103,24 @@ impl Bank {
     }
 
     fn check_message_for_nonce(&self, message: &SanitizedMessage) -> Option<TransactionAccount> {
-        message
-            .get_durable_nonce(self.feature_set.is_active(&nonce_must_be_writable::id()))
-            .and_then(|nonce_address| {
-                self.get_account_with_fixed_root(nonce_address)
-                    .map(|nonce_account| (*nonce_address, nonce_account))
-            })
-            .filter(|(_, nonce_account)| {
-                nonce_account::verify_nonce_account(nonce_account, message.recent_blockhash())
-            })
+        let nonce_address =
+            message.get_durable_nonce(self.feature_set.is_active(&nonce_must_be_writable::id()))?;
+        let nonce_account = self.get_account_with_fixed_root(nonce_address)?;
+        let nonce_data = nonce_account::get_nonce_data(&nonce_account)?;
+        if message.recent_blockhash() != &nonce_data.blockhash() {
+            return None;
+        }
+
+        if self
+            .feature_set
+            .is_active(&feature_set::nonce_must_be_authorized::ID)
+        {
+            message
+                .get_ix_signers(NONCED_TX_MARKER_IX_INDEX as usize)?
+                .find(|signer| *signer == &nonce_data.authority)?;
+        }
+
+        Some((*nonce_address, nonce_account))
```

Comment:
`.any(...)` instead of `.find`. Also the `?;` for validating authority feels like `?` abuse.

Alternatively
```rust
let signed_by_auhtority = !self
    .feature_set
    .is_active(&feature_set::nonce_must_be_authorized::ID)
    || message
        .get_ix_signers(NONCED_TX_MARKER_IX_INDEX as usize)?
        .any(|signer| signer == &nonce_data.authority);
signed_by_auhtority.then(|| (*nonce_address, nonce_account))
```